<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>Tutorials_PopGenomics3.md</title><link href='http://fonts.googleapis.com/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext' rel='stylesheet' type='text/css'><style type='text/css'>html, body {overflow-x: initial !important;}.CodeMirror { height: auto; }
.CodeMirror-scroll { overflow-y: hidden; overflow-x: auto; }
.CodeMirror-lines { padding: 4px 0px; }
.CodeMirror pre { }
.CodeMirror-scrollbar-filler, .CodeMirror-gutter-filler { background-color: white; }
.CodeMirror-gutters { border-right-width: 1px; border-right-style: solid; border-right-color: rgb(221, 221, 221); background-color: rgb(247, 247, 247); white-space: nowrap; }
.CodeMirror-linenumbers { }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.CodeMirror div.CodeMirror-cursor { border-left-width: 1px; border-left-style: solid; border-left-color: black; z-index: 3; }
.CodeMirror div.CodeMirror-secondarycursor { border-left-width: 1px; border-left-style: solid; border-left-color: silver; }
.CodeMirror.cm-keymap-fat-cursor div.CodeMirror-cursor { width: auto; border: 0px; background-color: rgb(119, 238, 119); z-index: 1; background-position: initial initial; background-repeat: initial initial; }
.CodeMirror div.CodeMirror-cursor.CodeMirror-overwrite { }
.cm-tab { display: inline-block; }
.cm-s-typora-default .cm-header, .cm-s-typora-default .cm-property { color: rgb(217, 79, 138); }
.cm-s-typora-default pre.cm-header1:not(.cm-atom) :not(.cm-overlay) { font-size: 2rem; line-height: 2rem; }
.cm-s-typora-default pre.cm-header2:not(.cm-atom) :not(.cm-overlay) { font-size: 1.4rem; line-height: 1.4rem; }
.cm-s-typora-default .cm-atom, .cm-s-typora-default .cm-number { color: rgb(149, 132, 134); }
.cm-s-typora-default .cm-table-row, .cm-s-typora-default .cm-block-start { font-family: monospace; }
.cm-s-typora-default .cm-comment, .cm-s-typora-default .cm-code { color: rgb(74, 90, 159); font-family: monospace; }
.cm-s-typora-default .cm-tag { color: rgb(169, 68, 66); }
.cm-s-typora-default .cm-string { color: rgb(126, 134, 169); }
.cm-s-typora-default .cm-link { color: rgb(196, 122, 15); text-decoration: underline; }
.cm-s-typora-default .cm-variable-2, .cm-s-typora-default .cm-variable-1 { color: inherit; }
.cm-s-typora-default .cm-overlay { font-size: 1rem; font-family: monospace; }
.CodeMirror.cm-s-typora-default div.CodeMirror-cursor { border-left-width: 3px; border-left-style: solid; border-left-color: rgb(228, 98, 154); }
.cm-s-typora-default .CodeMirror-activeline-background { left: -60px; right: -30px; background-color: rgba(204, 204, 204, 0.2); background-position: initial initial; background-repeat: initial initial; }
.cm-s-typora-default .CodeMirror-gutters { border-right-style: none; background-color: inherit; }
.cm-s-typora-default .cm-trailing-space-new-line::after, .cm-startspace::after, .cm-starttab .cm-tab::after { content: '•'; position: absolute; left: 0px; opacity: 0; font-family: LetterGothicStd, monospace; }
.os-windows .cm-startspace::after, .os-windows .cm-starttab .cm-tab::after { left: -0.1em; }
.cm-starttab .cm-tab::after { content: ' '; }
.cm-startspace, .cm-tab, .cm-starttab, .cm-trailing-space-a, .cm-trailing-space-b, .cm-trailing-space-new-line { font-family: monospace; position: relative; }
.cm-s-typora-default .cm-trailing-space-new-line::after { content: '↓'; opacity: 0.3; }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: black; }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-property { color: black; }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: blue; }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: bold; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: rgb(255, 0, 0); }
.cm-invalidchar { color: rgb(255, 0, 0); }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; background-position: inherit inherit; background-repeat: inherit inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { margin-bottom: -30px; margin-right: -30px; padding-bottom: 30px; padding-right: 30px; height: 100%; outline: none; position: relative; box-sizing: content-box; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-vscrollbar, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-gutter-filler { position: absolute; z-index: 6; display: none; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow-x: hidden; overflow-y: scroll; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow-y: hidden; overflow-x: scroll; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 30px; z-index: 3; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-top-left-radius: 0px; border-top-right-radius: 0px; border-bottom-right-radius: 0px; border-bottom-left-radius: 0px; border-width: 0px; background-color: transparent; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; word-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; background-position: initial initial; background-repeat: initial initial; }
.CodeMirror-wrap pre { word-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right-width: 30px; border-right-style: solid; border-right-color: transparent; width: -webkit-fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right-style: none; width: auto; }
.CodeMirror-linebackground { position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-widget { }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right-style: none; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.CodeMirror-selected { background-color: rgb(217, 217, 217); background-position: initial initial; background-repeat: initial initial; }
.CodeMirror-focused .CodeMirror-selected { background-color: rgb(215, 212, 240); background-position: initial initial; background-repeat: initial initial; }
.cm-searching { background-color: rgba(255, 255, 0, 0.4); background-position: initial initial; background-repeat: initial initial; }
.CodeMirror span { }
@media print { 
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}
.CodeMirror-lint-markers { width: 16px; }
.CodeMirror-lint-tooltip { background-color: infobackground; border: 1px solid black; border-top-left-radius: 4px; border-top-right-radius: 4px; border-bottom-right-radius: 4px; border-bottom-left-radius: 4px; color: infotext; font-family: monospace; overflow: hidden; padding: 2px 5px; position: fixed; white-space: pre-wrap; z-index: 10000; max-width: 600px; opacity: 0; -webkit-transition: opacity 0.4s; transition: opacity 0.4s; font-size: 0.8em; }
.CodeMirror-lint-mark-error, .CodeMirror-lint-mark-warning { background-position: 0% 100%; background-repeat: repeat no-repeat; }
.CodeMirror-lint-mark-error { background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAQAAAADCAYAAAC09K7GAAAAAXNSR0IArs4c6QAAAAZiS0dEAP8A/wD/oL2nkwAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9sJDw4cOCW1/KIAAAAZdEVYdENvbW1lbnQAQ3JlYXRlZCB3aXRoIEdJTVBXgQ4XAAAAHElEQVQI12NggIL/DAz/GdA5/xkY/qPKMDAwAADLZwf5rvm+LQAAAABJRU5ErkJggg==); }
.CodeMirror-lint-marker-error, .CodeMirror-lint-marker-warning { cursor: pointer; display: inline-block; height: 16px; width: 16px; vertical-align: middle; position: relative; background-position: 50% 50%; background-repeat: no-repeat no-repeat; }
.CodeMirror-lint-message-error, .CodeMirror-lint-message-warning { padding-left: 18px; background-position: 0% 0%; background-repeat: no-repeat no-repeat; }
.CodeMirror-lint-marker-error, .CodeMirror-lint-message-error { background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAAAHlBMVEW7AAC7AACxAAC7AAC7AAAAAAC4AAC5AAD///+7AAAUdclpAAAABnRSTlMXnORSiwCK0ZKSAAAATUlEQVR42mWPOQ7AQAgDuQLx/z8csYRmPRIFIwRGnosRrpamvkKi0FTIiMASR3hhKW+hAN6/tIWhu9PDWiTGNEkTtIOucA5Oyr9ckPgAWm0GPBog6v4AAAAASUVORK5CYII=); }
.CodeMirror-lint-marker-warning, .CodeMirror-lint-message-warning { background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAAANlBMVEX/uwDvrwD/uwD/uwD/uwD/uwD/uwD/uwD/uwD6twD/uwAAAADurwD2tQD7uAD+ugAAAAD/uwDhmeTRAAAADHRSTlMJ8mN1EYcbmiixgACm7WbuAAAAVklEQVR42n3PUQqAIBBFUU1LLc3u/jdbOJoW1P08DA9Gba8+YWJ6gNJoNYIBzAA2chBth5kLmG9YUoG0NHAUwFXwO9LuBQL1giCQb8gC9Oro2vp5rncCIY8L8uEx5ZkAAAAASUVORK5CYII=); }
.CodeMirror-lint-marker-multiple { background-image: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAcAAAAHCAMAAADzjKfhAAAACVBMVEUAAAAAAAC/v7914kyHAAAAAXRSTlMAQObYZgAAACNJREFUeNo1ioEJAAAIwmz/H90iFFSGJgFMe3gaLZ0od+9/AQZ0ADosbYraAAAAAElFTkSuQmCC); width: 100%; height: 100%; background-position: 100% 100%; background-repeat: no-repeat no-repeat; }


html { font-size: 14px; background-color: rgb(255, 255, 255); color: rgb(51, 51, 51); }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif; font-size: 1rem; line-height: 1.42857143; overflow-x: hidden; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; background-position: inherit inherit; background-repeat: inherit inherit; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { background-color: rgb(181, 214, 252); text-shadow: none; background-position: initial initial; background-repeat: initial initial; }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; word-wrap: break-word; position: relative; padding-bottom: 70px; white-space: pre-wrap; overflow-x: auto; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
@media screen and (max-width: 500px) { 
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
.typora-export #write { margin: 0px auto; }
#write > p:first-child, #write > ul:first-child, #write > ol:first-child, #write > pre:first-child, #write > blockquote:first-child, #write > div:first-child, #write > table:first-child { margin-top: 30px; }
#write li > table:first-child { margin-top: -20px; }
img { max-width: 100%; vertical-align: middle; }
input, button, select, textarea { color: inherit; font-family: inherit; font-size: inherit; font-style: inherit; font-variant: inherit; font-weight: inherit; line-height: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
::before, ::after, * { box-sizing: border-box; }
#write p, #write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write div, #write pre { width: inherit; }
#write p, #write h1, #write h2, #write h3, #write h4, #write h5, #write h6 { position: relative; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
p { -webkit-margin-before: 1rem; -webkit-margin-after: 1rem; -webkit-margin-start: 0px; -webkit-margin-end: 0px; }
.mathjax-block { margin-top: 0px; margin-bottom: 0px; -webkit-margin-before: 0rem; -webkit-margin-after: 0rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: bold; font-style: italic; }
a { cursor: pointer; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; margin: 4px 0px 0px; }
tr { page-break-inside: avoid; page-break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; page-break-inside: auto; text-align: left; }
table.md-table td { min-width: 80px; }
.CodeMirror-gutters { border-right-width: 0px; background-color: inherit; }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
pre { white-space: pre-wrap; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; page-break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background-image: inherit; background-size: inherit; background-attachment: inherit; background-origin: inherit; background-clip: inherit; background-color: inherit; position: relative !important; background-position: inherit inherit; background-repeat: inherit inherit; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
.md-fences .CodeMirror.CodeMirror-wrap { top: -1.6em; margin-bottom: -1.6em; }
.md-fences.mock-cm { white-space: pre-wrap; }
.show-fences-line-number .md-fences { padding-left: 0px; }
.show-fences-line-number .md-fences.mock-cm { padding-left: 40px; }
.footnotes { opacity: 0.8; font-size: 0.9rem; padding-top: 1em; padding-bottom: 1em; }
.footnotes + .footnotes { margin-top: -1em; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background-color: transparent; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; line-height: normal; font-weight: normal; text-align: left; box-sizing: content-box; direction: ltr; background-position: initial initial; background-repeat: initial initial; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li p, li .mathjax-block { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; }
@media print { 
  html, body { height: 100%; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  h1, h2, h3, h4, h5, h6 { page-break-after: avoid; orphans: 2; }
  p { orphans: 4; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 1cm; padding-right: 1cm; }
  .typora-export #write::after { height: 0px; }
  @page { margin: 20mm 0mm; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 2.86rem; white-space: pre-wrap; background-color: rgb(204, 204, 204); display: block; overflow-x: hidden; background-position: initial initial; background-repeat: initial initial; }
p .md-image:only-child { display: inline-block; width: 100%; text-align: center; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.mathjax-block { white-space: pre; overflow: hidden; width: 100%; }
p + .mathjax-block { margin-top: -1.143rem; }
.mathjax-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus { outline: none; box-shadow: none; }
.task-list { list-style-type: none; }
.task-list-item { position: relative; padding-left: 1em; }
.task-list-item input { position: absolute; top: 0px; left: 0px; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-top-left-radius: 10px; border-top-right-radius: 10px; border-bottom-right-radius: 10px; border-bottom-left-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc::after, .md-toc-content::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); text-decoration: none; }
.md-toc-inner:hover { }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: bold; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) { 
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: '.'; }
.md-tag { opacity: 0.5; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: monospace; }
code { text-align: left; }
h1 .md-tag, h2 .md-tag, h3 .md-tag, h4 .md-tag, h5 .md-tag, h6 .md-tag { font-weight: initial; opacity: 0.35; }
a.md-print-anchor { border: none !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: none !important; background-color: transparent !important; text-shadow: initial !important; background-position: initial initial !important; background-repeat: initial initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.mathjax-block .MathJax_SVG_Display { text-align: center; margin: 1em 0em; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: monospace; }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: normal; line-height: normal; zoom: 90%; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { -webkit-transition: none; transition: none; }


@include-when-export url(http://fonts.googleapis.com/css?family=Open+Sans:400italic,700italic,700,400&subset=latin,latin-ext);

/**
 * css forked from https://github.com/GitbookIO/gitbook 
 * www.gitbook.com
 * Apache License
 * https://github.com/GitbookIO/gitbook/blob/master/LICENSE
 **/

@font-face {
    font-family: 'Open Sans';
    font-style: normal;
    font-weight: normal;
    src: local('Open Sans Regular'),url('file:///Users/srkeller/Library/Application%20Support/abnerworks.Typora/themes/github/400.woff') format('woff')
}

@font-face {
    font-family: 'Open Sans';
    font-style: italic;
    font-weight: normal;
    src: local('Open Sans Italic'),url('file:///Users/srkeller/Library/Application%20Support/abnerworks.Typora/themes/github/400i.woff') format('woff')
}

@font-face {
    font-family: 'Open Sans';
    font-style: normal;
    font-weight: bold;
    src: local('Open Sans Bold'),url('file:///Users/srkeller/Library/Application%20Support/abnerworks.Typora/themes/github/700.woff') format('woff')
}

@font-face {
    font-family: 'Open Sans';
    font-style: italic;
    font-weight: bold;
    src: local('Open Sans Bold Italic'),url('file:///Users/srkeller/Library/Application%20Support/abnerworks.Typora/themes/github/700i.woff') format('woff')
}

html {
    font-size: 16px;
}

body {
    font-family: "Open Sans","Clear Sans","Helvetica Neue",Helvetica,Arial,sans-serif;
    color: rgb(51, 51, 51);
    line-height: 1.6;
}

#write{
    max-width: 860px;
  	margin: 0 auto;
  	padding: 20px 30px 40px 30px;
	padding-top: 20px;
    padding-bottom: 100px;
}
#write > ul:first-child,
#write > ol:first-child{
    margin-top: 30px;
}

body > *:first-child {
    margin-top: 0 !important;
}
body > *:last-child {
    margin-bottom: 0 !important;
}
a {
    color: #4183C4;
}
h1,
h2,
h3,
h4,
h5,
h6 {
    position: relative;
    margin-top: 1rem;
    margin-bottom: 1rem;
    font-weight: bold;
    line-height: 1.4;
    cursor: text;
}
h1:hover a.anchor,
h2:hover a.anchor,
h3:hover a.anchor,
h4:hover a.anchor,
h5:hover a.anchor,
h6:hover a.anchor {
    /*background: url("file:///Users/srkeller/Library/Application%20Support/images/modules/styleguide/para.png") no-repeat 10px center;*/
    text-decoration: none;
}
h1 tt,
h1 code {
    font-size: inherit;
}
h2 tt,
h2 code {
    font-size: inherit;
}
h3 tt,
h3 code {
    font-size: inherit;
}
h4 tt,
h4 code {
    font-size: inherit;
}
h5 tt,
h5 code {
    font-size: inherit;
}
h6 tt,
h6 code {
    font-size: inherit;
}
h1 {
    padding-bottom: .3em;
    font-size: 2.25em;
    line-height: 1.2;
    border-bottom: 1px solid #eee;
}
h2 {
   padding-bottom: .3em;
    font-size: 1.75em;
    line-height: 1.225;
    border-bottom: 1px solid #eee;
}
h3 {
    font-size: 1.5em;
    line-height: 1.43;
}
h4 {
    font-size: 1.25em;
}
h5 {
    font-size: 1em;
}
h6 {
   font-size: 1em;
    color: #777;
}
p,
blockquote,
ul,
ol,
dl,
table{
    margin: 0.8em 0;
}
li>ol,
li>ul {
    margin: 0 0;
}
hr {
    height: 4px;
    padding: 0;
    margin: 16px 0;
    background-color: #e7e7e7;
    border: 0 none;
    overflow: hidden;
    box-sizing: content-box;
    border-bottom: 1px solid #ddd;
}

body > h2:first-child {
    margin-top: 0;
    padding-top: 0;
}
body > h1:first-child {
    margin-top: 0;
    padding-top: 0;
}
body > h1:first-child + h2 {
    margin-top: 0;
    padding-top: 0;
}
body > h3:first-child,
body > h4:first-child,
body > h5:first-child,
body > h6:first-child {
    margin-top: 0;
    padding-top: 0;
}
a:first-child h1,
a:first-child h2,
a:first-child h3,
a:first-child h4,
a:first-child h5,
a:first-child h6 {
    margin-top: 0;
    padding-top: 0;
}
h1 p,
h2 p,
h3 p,
h4 p,
h5 p,
h6 p {
    margin-top: 0;
}
li p.first {
    display: inline-block;
}
ul,
ol {
    padding-left: 30px;
}
ul:first-child,
ol:first-child {
    margin-top: 0;
}
ul:last-child,
ol:last-child {
    margin-bottom: 0;
}
blockquote {
    border-left: 4px solid #dddddd;
    padding: 0 15px;
    color: #777777;
}
blockquote blockquote {
    padding-right: 0;
}
table {
    padding: 0;
    word-break: initial;
}
table tr {
    border-top: 1px solid #cccccc;
    background-color: white;
    margin: 0;
    padding: 0;
}
table tr:nth-child(2n) {
    background-color: #f8f8f8;
}
table tr th {
    font-weight: bold;
    border: 1px solid #cccccc;
    text-align: left;
    margin: 0;
    padding: 6px 13px;
}
table tr td {
    border: 1px solid #cccccc;
    text-align: left;
    margin: 0;
    padding: 6px 13px;
}
table tr th:first-child,
table tr td:first-child {
    margin-top: 0;
}
table tr th:last-child,
table tr td:last-child {
    margin-bottom: 0;
}

.CodeMirror-gutters {
    border-right: 1px solid #ddd;
}

.md-fences,
code,
tt {
    border: 1px solid #ddd;
    background-color: #f8f8f8;
    border-radius: 3px;
    padding: 0;
    font-family: Consolas, "Liberation Mono", Courier, monospace;
    padding: 2px 4px 0px 4px;
    font-size: 0.9em;
}

.md-fences {
    margin-bottom: 15px;
    margin-top: 15px;
    padding: 0.2em 1em;
    padding-top: 8px;
    padding-bottom: 6px;
}
.task-list{
	padding-left: 0;
}

.task-list-item {
	padding-left:32px;
}

.task-list-item input {
  top: 3px;
  left: 8px;
}

@media screen and (min-width: 914px) {
    /*body {
        width: 854px;
        margin: 0 auto;
    }*/
}
@media print {
    html {
        font-size: 13px;
    }
    table,
    pre {
        page-break-inside: avoid;
    }
    pre {
        word-wrap: break-word;
    }
}

.md-fences {
	background-color: #f8f8f8;
}
#write pre.md-meta-block {
	padding: 1rem;
    font-size: 85%;
    line-height: 1.45;
    background-color: #f7f7f7;
    border: 0;
    border-radius: 3px;
    color: #777777;
    margin-top: 0 !important;
}

.mathjax-block>.code-tooltip {
	bottom: .375rem;
}

#write>h3.md-focus:before{
	left: -1.5625rem;
	top: .375rem;
}
#write>h4.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h5.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
#write>h6.md-focus:before{
	left: -1.5625rem;
	top: .285714286rem;
}
.md-image>.md-meta {
    border: 1px solid #ddd;
    border-radius: 3px;
    font-family: Consolas, "Liberation Mono", Courier, monospace;
    padding: 2px 4px 0px 4px;
    font-size: 0.9em;
    color: inherit;
}

.md-tag{
	color: inherit;
}

.md-toc { 
    margin-top:20px;
    padding-bottom:20px;
}

#typora-quick-open {
    border: 1px solid #ddd;
    background-color: #f8f8f8;
}

#typora-quick-open-item {
    background-color: #FAFAFA;
    border-color: #FEFEFE #e5e5e5 #e5e5e5 #eee;
    border-style: solid;
    border-width: 1px;
}

#md-notification:before {
    top: 10px;
}

/** focus mode */
.on-focus-mode blockquote {
    border-left-color: rgba(85, 85, 85, 0.12);
}

header, .context-menu, .megamenu-content, footer{
    font-family: "Segoe UI", "Arial", sans-serif;
}


</style>
</head>
<body class='typora-export' >
<div  id='write'  class = 'is-mac'><h1><a name='header-c391' class='md-header-anchor '></a>P/BIO381 Tutorials</h1><h2><a name='header-c393' class='md-header-anchor '></a>Population Genomics 3: Finishing allele frequency and diversity calculations</h2><h3><a name='header-c394' class='md-header-anchor '></a>March 20, 2017</h3><p>Today, we&#39;ll finish up our calculations of allele frequencies and nucleotide diversity in the SSW data, before moving onto testing if there&#39;s population structure (next session). </p><p>First, recall that our previous SNP vcf file had 22 of 24 individuals in it. I found the missing 2 individuals (!) and called SNPs for all 24 individuals now using <strong>reads2snps</strong>. </p><p><strong>PATH TO THE FINAL VCF DATA (all 24 INDS):</strong> </p><pre class='md-fences mock-cm' style='display:block;position:relative'>/data/_project_data/snps/reads2snps/SSW_by24inds.txt.vcf.gz</pre><ul><li><em>Use VCFTools to filter genotypes and save in your home directory in gzipped format:</em></li></ul><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap  "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre><span style="padding-right: 0.1px;"><span class="cm-def">$ cd</span> /data/_project_data/snps/reads2snps/</span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ vcftools</span> <span class="cm-attribute">--gzvcf</span> SSW_by24inds.txt.vcf.gz <span class="cm-attribute">--min-alleles</span> <span class="cm-number">2</span> <span class="cm-attribute">--max-alleles</span> <span class="cm-number">2</span> <span class="cm-attribute">--maf</span> <span class="cm-number">0</span>.02 <span class="cm-attribute">--max-missing</span> <span class="cm-number">0</span>.8 <span class="cm-attribute">--recode</span> <span class="cm-attribute">--out</span> ~/SSW_all_biallelic.MAF0.02.Miss0.8  </span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ cd</span> ~/</span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ gzip</span> SSW_all_biallelic.MAF0.02.Miss0.8.recode.vcf</span></pre></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 110px;"></div></div></div></pre><hr /><h2><a name='header-c408' class='md-header-anchor '></a>Estimate allele frequencies in H and S:</h2><p><strong>Let&#39;s compare the SNP frequencies for all loci between Healthy and Sick animals</strong>. Perhaps there are some loci that contribute to a difference in pathogen susceptibility, which could be identified this way. Let&#39;s take a look.</p><p>First, we need to re-create our text files containing the individual ID&#39;s for just Sick (and later, just Healthy) individuals. These meta-data are in the following file:</p><pre class='md-fences mock-cm' style='display:block;position:relative'>/data/project_data/snps/reads2snps/ssw_healthloc.txt</pre><p>Remember how to parse this file on the command line to get <em>just the <strong>healthy</strong> individual IDs</em>? Here&#39;s the strategy at the command-line:</p><ul><li>Use <strong>grep</strong> to match rows of data for disease trajectory</li><li>Pipe the results to <strong>cut</strong> to grab just the 1st column corresponding to sample ID (-f1). </li><li>Save this to your home directory and name it: <strong>&quot;H_SampleIDs.txt&quot;</strong></li></ul><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ grep</span> <span class="cm-string">"HH"</span> ssw_healthloc.txt | <span class="cm-builtin">cut</span> <span class="cm-attribute">-f1</span> &gt;~/H_SampleIDs.txt</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><ul><li>Confirm  that there are 8 individuals in your output file.</li><li>Do the same for Sick individuals, saving to your home directory as: <strong>&quot;S_SampleIDs.txt&quot;</strong>. The <strong>grep</strong> command here is a little different, since we want to match <em>either</em> HS <em>or</em> SS.  The &quot;\|&quot; part of the match tells  <strong>grep</strong> match HS <em>OR</em> SS</li></ul><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ grep</span> <span class="cm-string">"HS\|SS"</span> ssw_healthloc.txt | <span class="cm-builtin">cut</span> <span class="cm-attribute">-f1</span> &gt;~/S_SampleIDs.txt</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><ul><li>Confirm that there are 14 individuals in your output file</li></ul><p>Now call VCFtools on your filtered gzipped vcf file saved in your home directory to calculate allele frequencies for each group. This will require 2 separate calls to VCFtools.</p><p><strong>Allele Frequencies between Healthy and Sick individuals:</strong></p><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre><span style="padding-right: 0.1px;"><span class="cm-def">$ cd</span> ~/path to your filtered vcf file <span class="cm-keyword">in</span> your home directory...</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ vcftools</span> <span class="cm-attribute">--gzvcf</span> SSW_all_biallelic.MAF0.02.Miss0.8.recode.vcf.gz <span class="cm-attribute">--freq2</span> <span class="cm-attribute">--keep</span> H_SampleIDs.txt <span class="cm-attribute">--out</span> H_AlleleFreqs</span></pre></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 66px;"></div></div></div></pre><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre><span style="padding-right: 0.1px;"><span class="cm-def">$ vcftools</span> <span class="cm-attribute">--gzvcf</span> SSW_all_biallelic.MAF0.02.Miss0.8.recode.vcf.gz <span class="cm-attribute">--freq2</span> <span class="cm-attribute">--keep</span> S_SampleIDs.txt <span class="cm-attribute">--out</span> S_AlleleFreqs</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 44px;"></div></div></div></pre><p></p><p>Let&#39;s also calculate Wright&#39;s Fst between H and S groups, which standardizes allele frequency difference based on the mean within groups. </p><p><strong>Fst between Healthy and Sick individuals:</strong></p><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre><span style="padding-right: 0.1px;"><span class="cm-def">$ vcftools</span> <span class="cm-attribute">--gzvcf</span> SSW_all_biallelic.MAF0.02.Miss0.8.recode.vcf.gz <span class="cm-attribute">--weir-fst-pop</span> H_SampleIDs.txt <span class="cm-attribute">--weir-fst-pop</span> S_SampleIDs.txt <span class="cm-attribute">--out</span> HvS_Fst</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 44px;"></div></div></div></pre><p></p><p>Now, we can import these datasets into <strong>R</strong> and make some plots to examine how the diversity and differentiation varies in our dataset:</p><ol><li><p>Download all 3 new results files to your laptop using <strong>scp</strong> or <strong>Fetch</strong> [MacOS] or <strong>WinScp</strong> [PC]</p></li><li><p>Open the <strong>H_AlleleFreqs.frq</strong> file in a text editor and edit the header line as follows:</p><ol><li>DELETE:   {Freq}</li><li>REPLACE with:   H_REF <tab> H_ALT</li><li>Do the same for the <strong>S_AlleleFreqs.frq</strong> file...</li></ol></li><li><p>Open <strong>R</strong>, paste the following into an R script and work through it:</p></li></ol><pre class="md-fences md-end-block" lang="r"> <div class="CodeMirror cm-s-inner CodeMirror-wrap   "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Set your working directory to where you downloaded your results files:</span></span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">setwd</span>(<span class="cm-string">"~/github/PBIO381_srkeller_labnotebook/results/"</span>)</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># List the files in this directory -- you should see your results output from VCFTools if the download was successful</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">list.files</span>()</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Let's do the allele freq comparisons first:</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">H_freq</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">read.table</span>(<span class="cm-string">"H_AlleleFreqs.frq"</span>, <span class="cm-variable">header</span><span class="cm-arg-is">=</span><span class="cm-variable">T</span>)</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">S_freq</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">read.table</span>(<span class="cm-string">"S_AlleleFreqs.frq"</span>, <span class="cm-variable">header</span><span class="cm-arg-is">=</span><span class="cm-variable">T</span>)</span></pre><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Since these files have identical numbers of SNPs in the exact same order, we can concatenate them together into one large dataframe:</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">All_freq</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">merge</span>(<span class="cm-variable">H_freq</span>, <span class="cm-variable">S_freq</span>, <span class="cm-variable">by</span><span class="cm-arg-is">=</span><span class="cm-variable">c</span>(<span class="cm-string">"CHROM"</span>, <span class="cm-string">"POS"</span>))</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Check the results of your merge to make sure things look OK</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">str</span>(<span class="cm-variable">All_freq</span>)</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">head</span>(<span class="cm-variable">All_freq</span>)</span></pre><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Looks good, now let's calculate the difference in minor allele frequency at each SNP and plot as a histogram</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">All_freq</span><span class="cm-dollar">$</span><span class="cm-variable">diff</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">All_freq</span><span class="cm-dollar">$</span><span class="cm-variable">H_ALT</span> <span class="cm-operator">-</span> <span class="cm-variable">All_freq</span><span class="cm-dollar">$</span><span class="cm-variable">S_ALT</span></span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">hist</span>(<span class="cm-variable">All_freq</span><span class="cm-dollar">$</span><span class="cm-variable">diff</span>, <span class="cm-variable">breaks</span><span class="cm-arg-is">=</span><span class="cm-number">50</span>, <span class="cm-variable">col</span><span class="cm-arg-is">=</span><span class="cm-string">"red"</span>, <span class="cm-string">"Allele frequency difference (H-S)"</span>)</span></pre><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Looks like most loci show little difference (i.e., mostly drift), but perhaps a few show very large differences between healthy and sick</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># How do these highly divergent frequenices compare to Fst at the same SNPs?</span></span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">fst</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">read.table</span>(<span class="cm-string">"HvS_Fst.weir.fst"</span>, <span class="cm-variable">header</span><span class="cm-arg-is">=</span><span class="cm-variable">T</span>)</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">All_freq.fst</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">merge</span>(<span class="cm-variable">All_freq</span>, <span class="cm-variable">fst</span>, <span class="cm-variable">by</span><span class="cm-arg-is">=</span><span class="cm-variable">c</span>(<span class="cm-string">"CHROM"</span>, <span class="cm-string">"POS"</span>))</span></pre><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">plot</span>(<span class="cm-variable">All_freq.fst</span><span class="cm-dollar">$</span><span class="cm-variable">diff</span>, <span class="cm-variable">All_freq.fst</span><span class="cm-dollar">$</span><span class="cm-variable">WEIR_AND_COCKERHAM_FST</span>, <span class="cm-variable">xlab</span><span class="cm-arg-is">=</span><span class="cm-string">"Allele frequency difference (H-S)"</span>, <span class="cm-variable">ylab</span><span class="cm-arg-is">=</span><span class="cm-string">"Fst"</span>, <span class="cm-variable">main</span><span class="cm-arg-is">=</span><span class="cm-string">"Healthy vs. Sick SNP divergence"</span>)</span></pre><pre class=""><span style="padding-right: 0.1px;"><span>​</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Which are the genes that are showing the highest divergence between Healthy and Sick?</span></span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">All_freq.fst</span>[<span class="cm-variable">which</span>(<span class="cm-variable">All_freq.fst</span><span class="cm-dollar">$</span><span class="cm-variable">WEIR_AND_COCKERHAM_FST</span><span class="cm-operator">&gt;</span><span class="cm-number">0.2</span>),]</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 836px;"></div></div></div></pre><p></p><hr /><h3><a name='header-c480' class='md-header-anchor '></a>Comparing Sea Star nucleotide diversity and piN/piS to the sample of metazoans that Romiguier et al. (2014) report.</h3><p>Romiguier et al. report on some very intriguing associations between species life history traits, nucleotide diversity at synonymous sites (piS), and the ratio of piN/piS (where piN is nucleotide diversity at nonsynonymous site). </p><p><img src='http://www.nature.com/nature/journal/v515/n7526/images/nature13685-f2.jpg' alt='Romiguier_Figure2' /></p><p></p><p>What do our sea star data have to say about this? Or more importantly: <strong><em>where do sea stars fall on the genomic diversity ~ life history continuum?</em></strong></p><p>Estimating piS and piN on the entire dataset will take some time. Let&#39;s try and set this up at the end of the day and let it run. </p><p>We&#39;ll use the <strong>piNpiS</strong> script from Gayral et al. (2013) to run this. We only need a single input file, which is a FASTA formatted sequence file that is output from <strong>reads2snps</strong>, and we&#39;ll save the output to our home directories:</p><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ cd</span> /data/project_data/snps/reads2snps</span></pre></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ </span>/data/popgen/dNdSpiNpiS_1.0 <span class="cm-attribute">-alignment_file</span><span class="cm-operator">=</span>SSW_by24inds.txt.fas <span class="cm-attribute">-ingroup</span><span class="cm-operator">=</span>sp <span class="cm-attribute">-out</span><span class="cm-operator">=</span>~/dNdSpiNpiS_output</span></pre></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 66px;"></div></div></div></pre><p></p><p>While we wait for that to chug along (it&#39;ll calculate confidence intervals from 10,000 bootstraps…which takes ~ 5 hours on our data), we can look at the summary output from the smaller VCF file run previously on just 1 sample library per individual</p><ul><li><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap   "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-def">$ cat</span> SSW_bamlist.txt.sum</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre></li><li><p>Record what we got for our sea stars:</p><ul><li>piS: <strong>__</strong>___</li><li>piN: <strong>__</strong>___</li><li>ave. piN/piS: <strong>__</strong>___</li></ul></li></ul><p>To compare the mean values across genes to Romiguier&#39;s data, we need to get their estimates and combine them with our estimates. I&#39;ve downloaded Romiguier&#39;s Table S3 to our server and saved as a common separated (.csv) file.</p><pre class="md-fences md-end-block" lang="bash"> <div class="CodeMirror cm-s-inner CodeMirror-wrap"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;">/data/project_data/snps/reads2snps/Romiguier_nature13685-s3.csv</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 22px;"></div></div></div></pre><p>Download this file to your laptop and then import it into <strong>R</strong>:</p><pre class="md-fences md-end-block" lang="r"> <div class="CodeMirror cm-s-inner CodeMirror-wrap      "><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 4px;"></div><div class="CodeMirror-hscrollbar" style="left: 0px;"><div style="height: 100%; min-height: 1px; width: 0px;"></div></div><div class="CodeMirror-vscrollbar"><div style="min-width: 1px; height: 0px;"></div></div><div class="CodeMirror-scrollbar-filler"></div><div class="CodeMirror-gutter-filler"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; min-height: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines"><div style="position: relative; outline: none;"><div class="CodeMirror-measure"><div style="width: 50px; height: 50px; overflow-x: scroll;"></div></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Set your working directory to where you downloaded your file:</span></span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">setwd</span>(<span class="cm-string">"~/github/PBIO381_srkeller_labnotebook/data/"</span>)</span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><div class="" style="position: relative;"><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># List the files in this directory</span></span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">list.files</span>()</span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><div class="" style="position: relative;"><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># Read in the Romiguier data:</span></span></pre></div><div class="" style="position: relative;"><pre><span style="padding-right: 0.1px;"><span class="cm-variable">Rom</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">read.csv</span>(<span class="cm-string">"Romiguier_nature13685-s3.csv"</span>, <span class="cm-variable">header</span><span class="cm-arg-is">=</span><span class="cm-variable">T</span>)</span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><div class="" style="position: relative;"><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># Import OK?</span></span></pre></div><div class="" style="position: relative;"><pre><span style="padding-right: 0.1px;"><span class="cm-variable">str</span>(<span class="cm-variable">Rom</span>) <span class="cm-comment"># shows the structure of the data</span></span></pre></div><pre><span style="padding-right: 0.1px;"><span class="cm-variable">head</span>(<span class="cm-variable">Rom</span>)</span></pre><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># Looks good; now let's look at how the strength of purifying selection (piN/piS) compares to the size of Ne (piS)</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">plot</span>(<span class="cm-variable">log</span>(<span class="cm-variable">Rom</span><span class="cm-dollar">$</span><span class="cm-variable">piS</span>), <span class="cm-variable">log</span>(<span class="cm-variable">Rom</span><span class="cm-dollar">$</span><span class="cm-variable">piNpiS</span>), <span class="cm-variable">pch</span><span class="cm-arg-is">=</span><span class="cm-number">21</span>, <span class="cm-variable">bg</span><span class="cm-arg-is">=</span><span class="cm-string">"blue"</span>, <span class="cm-variable">xlab</span><span class="cm-arg-is">=</span><span class="cm-string">"log Synonymous Nucleotide Diversity (piS)"</span>, <span class="cm-variable">ylab</span><span class="cm-arg-is">=</span><span class="cm-string">"log Ratio of Nonysn to Syn Diversity (piN/piS)"</span>, <span class="cm-variable">main</span><span class="cm-arg-is">=</span><span class="cm-string">"Purifying Selection vs. Effective Population Size"</span>)</span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># Now let's add in our estimates from the SSW data</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">points</span>(<span class="cm-variable">log</span>(<span class="cm-number">0.00585312</span>), <span class="cm-variable">log</span>(<span class="cm-number">0.264041</span>), <span class="cm-variable">pch</span><span class="cm-arg-is">=</span><span class="cm-number">24</span>, <span class="cm-variable">cex</span><span class="cm-arg-is">=</span><span class="cm-number">1.5</span>, <span class="cm-variable">bg</span><span class="cm-arg-is">=</span><span class="cm-string">"red"</span>) <span class="cm-comment"># adds our SSW points to the existing plot and gives them a different symbol</span></span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># We can add the regression line to the plot to see how far off the SSW estimates are from expectation</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">reg</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">lm</span>(<span class="cm-variable">log</span>(<span class="cm-variable">Rom</span><span class="cm-dollar">$</span><span class="cm-variable">piNpiS</span>) <span class="cm-operator">~</span> <span class="cm-variable">log</span>(<span class="cm-variable">Rom</span><span class="cm-dollar">$</span><span class="cm-variable">piS</span>)) <span class="cm-comment"># Fits a linear regression</span></span></pre></div><div class="" style="position: relative;"><pre><span style="padding-right: 0.1px;"><span class="cm-variable">abline</span>(<span class="cm-variable">reg</span>) <span class="cm-comment"># adds the line to the plot</span></span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># What about highlighting the other echinoderms in the dataset...do our seastars behave similarly?</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">echino</span> <span class="cm-arrow">&lt;-</span> <span class="cm-variable">Rom</span>[<span class="cm-variable">which</span>(<span class="cm-variable">Rom</span><span class="cm-dollar">$</span><span class="cm-variable">Phylum</span><span class="cm-arg-is">==</span><span class="cm-string">"Echinodermata"</span>),] <span class="cm-comment"># subsets the data</span></span></pre></div><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">points</span>(<span class="cm-variable">log</span>(<span class="cm-variable">echino</span><span class="cm-dollar">$</span><span class="cm-variable">piS</span>), <span class="cm-variable">log</span>(<span class="cm-variable">echino</span><span class="cm-dollar">$</span><span class="cm-variable">piNpiS</span>), <span class="cm-variable">pch</span><span class="cm-arg-is">=</span><span class="cm-number">21</span>, <span class="cm-variable">bg</span><span class="cm-arg-is">=</span><span class="cm-string">"red"</span>) <span class="cm-comment"># adds the points</span></span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><pre><span style="padding-right: 0.1px;"><span class="cm-comment"># Lastly, let's add a legend:</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-variable">legend</span>(<span class="cm-string">"bottomleft"</span>, <span class="cm-variable">cex</span><span class="cm-arg-is">=</span><span class="cm-number">1</span>, <span class="cm-variable">legend</span><span class="cm-arg-is">=</span><span class="cm-variable">c</span>(<span class="cm-string">"Metazoans"</span>, <span class="cm-string">"Echinoderms"</span>, <span class="cm-string">"P. ochraceus"</span>), <span class="cm-variable">pch</span><span class="cm-arg-is">=</span><span class="cm-variable">c</span>(<span class="cm-number">21</span>,<span class="cm-number">21</span>,<span class="cm-number">24</span>), <span class="cm-variable">col</span><span class="cm-arg-is">=</span><span class="cm-variable">c</span>(<span class="cm-string">"blue"</span>, <span class="cm-string">"red"</span>, <span class="cm-string">"red"</span>))</span></pre></div><pre><span style="padding-right: 0.1px;"><span>​</span></span></pre><div class="" style="position: relative;"><pre class=""><span style="padding-right: 0.1px;"><span class="cm-comment"># Pisaster seems to be in a group with other echinoderms that have relaxed purifying selection given their Ne...Interesting! Perhaps they went through a common demographic shift together, such as an historical bottleneck.</span></span></pre></div></div></div></div></div></div><div style="position: absolute; height: 30px; width: 1px; top: 0px;"></div><div class="CodeMirror-gutters" style="display: none; height: 880px;"></div></div></div></pre><p></p><p></p></div>
</body>
</html>